core/src/main/java/io/github/tcdl/Consumer.java

```diff
@@ -53,6 +55,7 @@ public Consumer(ConsumerAdapter rawAdapter, String topic, MsbConfigurations msbC
         this.clock = clock;
         this.channelMonitorAgent = channelMonitorAgent;
         this.validator = validator;
+        this.collectorManager = new CollectorManager(topic, channelManager);
```

Comment:
Also I think there's a bug here. collectorManager should be invoked in handleRawMessage. Shouldn't it?